airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/objects/impl/ConnectorSpecificationAdapter.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.protocol.objects.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.protocol.objects.ConnectorSpecification;
+import io.airbyte.commons.protocol.objects.DestinationSyncMode;
+import io.airbyte.protocol.models.AdvancedAuth;
+import io.airbyte.protocol.models.AuthSpecification;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+
+@EqualsAndHashCode
+public class ConnectorSpecificationAdapter implements ConnectorSpecification {
```

Comment:
also, how sketchy is it to have the generated protocol models directly implement the interfaces? I.e.
* interface declares `default Whatever getWhatever() { return null; }`
* and the generated classes look like `public class AirbyteMessage implements io.airbyte.commons.protocol.objects.AirbyteMessage`

then:
* adding new fields is still easy (just add a new `default` method to the interface)
* deleting fields is still easy (mark the interface method as deprecated, or just delete it entirely)
* renaming fields is... probably fine? (`default Whatever getWhatever() { return getOldWhatever(); }`)

(I forget if the issue was that there's a technical problem, or that the code generator just doesn't support this - if the latter, we could maaaaybe do something similar to https://github.com/LiveRamp/extravagance/ and edit the generated code)